compiler/AST/iterator.cpp

```diff
@@ -538,8 +559,10 @@ void setIteratorRecordShape(CallExpr* call) {
   INT_ASSERT(ir->type->symbol->hasFlag(FLAG_ITERATOR_RECORD));
   Symbol* shapeSpec = toSymExpr(call->get(2))->symbol();
   Symbol* fromForLoop = toSymExpr(call->get(3))->symbol();
+  Symbol* fromForeachLoop = toSymExpr(call->get(4))->symbol();
   CallExpr* shapeCall = setIteratorRecordShape(call, ir, shapeSpec,
-                          getSymbolImmediate(fromForLoop)->bool_value());
+                          getSymbolImmediate(fromForLoop)->bool_value(),
+                          getSymbolImmediate(fromForeachLoop)->bool_value());
```

Comment:
This is where things _might_ get a bit tricky, but I guess you can pick an `int_value` from an immediate that represents an enum value?